### PR TITLE
Fix TS compilation

### PR DESCRIPTION
**Pull request by markdboyd**
Apply fixes from #1988 to feature branch

### CODE REVIEW

**Review by botanical — COMMENTED** [review]
Did you intentionally leave out `-no-scripts` on [bamboo/deploy-dev-integration-test-stack.sh](https://github.com/nasa/cumulus/blob/fix-tsc-compilation/bamboo/deploy-dev-integration-test-stack.sh#L31)?

**Comment by markdboyd** [comment]
@botanical Yeah, by default [`lerna bootstrap`](https://github.com/lerna/lerna/tree/main/commands/bootstrap) runs the `prepare` and `prepublish` scripts in all of your Lerna packages. In most cases in CI, we don't want this, because our `prepare` scripts are set up to run `tsc` which compiles Typescript, but in CI we only want to do that once we generating a TS build cache and not in each step. 

However, there is one case where we want to run a `prepublish` step for the `ingest` module to generate a file needed for deployment. So I'm running `ci:bootstrap` once pre-deployment to at least run that step. 

9983bdc740f47e47f9e8c51e357951fb015cae2b

File: package.json [hunk 1]
```diff
@@ -25,14 +25,14 @@
...
-    "ci:bootstrap": "lerna bootstrap --no-ci --force-local",
+    "bootstrap": "lerna bootstrap --no-ci --force-local && npm run package",
+    "bootstrap-no-build": "lerna bootstrap --no-ci --force-local",
+    "bootstrap-no-build-no-scripts": "lerna bootstrap --no-ci --force-local --ignore-scripts",
+    "bootstrap-no-build-no-concurrency": "lerna bootstrap --no-ci --force-local --concurrency 1",
+    "bootstrap-no-build-quiet": "lerna bootstrap --no-ci --force-local --loglevel=error",
+    "ci:bootstrap": "lerna bootstrap --no-ci --force-local --ignore-scripts && lerna run prepublish",
     "ci:bootstrap-no-scripts": "lerna bootstrap --no-ci --force-local --ignore-scripts",
-    "ci:bootstrap-no-scripts-quiet": "lerna bootstrap --no-ci --force-local --loglevel=error",
+    "ci:bootstrap-no-scripts-quiet": "lerna bootstrap --no-ci --force-local --ignore-scripts --loglevel=error",
```
Comment (Jkovarik) [inline package.json:35]: This is the change from #1991

File: package.json [hunk 2]
```diff
@@ -25,14 +25,14 @@
...
     "tsc:listEmittedFiles": "lerna run tsc:listEmittedFiles --stream --no-prefix",
     "package": "lerna run package",
-    "bootstrap": "lerna bootstrap --no-ci --force-local --ignore-prepublish && npm run tsc && npm run package",
-    "bootstrap-no-build": "lerna bootstrap --no-ci --force-local && npm run tsc",
-    "bootstrap-no-build-no-scripts": "lerna bootstrap --no-ci --force-local --ignore-scripts && npm run tsc",
-    "bootstrap-no-build-no-concurrency": "lerna bootstrap --no-ci --force-local --concurrency 1 && npm run tsc",
-    "bootstrap-no-build-quiet": "lerna bootstrap --no-ci --force-local --loglevel=error && npm run tsc",
-    "ci:bootstrap": "lerna bootstrap --no-ci --force-local",
+    "bootstrap": "lerna bootstrap --no-ci --force-local && npm run package",
+    "bootstrap-no-build": "lerna bootstrap --no-ci --force-local",
```
Comment (Jkovarik) [inline package.json:29]: I'm feeling vague unease over the use of the term 'build' here.    That's not probably your work, just reflection on my part.   

Just for clarity - "no build" means following this PR: 

Do not package/webpack/etc but *do* compile typescript.

**Review by Jkovarik — APPROVED** [review]
:100: This looks good.    I've confirmed it works locally, and will defer to bamboo for a functional CI test. 

I left a couple of non-actionable comments.

**Pull request merged**